### PR TITLE
fix: reappearing tooltip after moving mouse within boundaries of triggered button

### PR DESCRIPTION
**Pull request by pudek357**
[SPT-1531](https://toptal-core.atlassian.net/browse/SPT-1531)

### Description
To prevent reappearing tooltip while moving mouse pointer within boundaries of triggered button, `onMouseLeave` has to be added.

### Note regaring "cypress-real-events" usage
There is no way to have "real hover" with cypress, please see cypress [docs](https://docs.cypress.io/api/commands/hover#See-also).

### How to test
- Tooltip -> Trigger section -> "hover" version
- Click on button to hide tooltip
- Try to move mouse over the button
- Tooltip shouldn't reapper again

### Screenshots
| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|  |  |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Comment by pudek357** [comment]
@toptal-bot run package:alpha-release

File: cypress/component/Tooltip.spec.tsx [hunk 1]
```diff
@@ -261,6 +275,35 @@ describe('Tooltip', () => {
     cy.get('body').happoScreenshot()
   })
 
+  it('renders on hover, and hides on click', () => {
```
Comment (pudek357) [inline cypress/component/Tooltip.spec.tsx:278]: It seems that this test on the local environment works and it does not pass on github

**Comment by pudek357** [comment]
@toptal-bot run package:alpha-release

**Review by teimurjan — APPROVED** [review]
Great job! 👍

File: cypress/component/Tooltip.spec.tsx (hunk 1 above)
Comment (pudek357) [reply cypress/component/Tooltip.spec.tsx:278]: Problem solved in: https://github.com/toptal/picasso/pull/2069/commits/1be76e4b1254d8a85649073413ab88a54ed73796

**Review by denieler — APPROVED** [review]

**Pull request merged**